v8.go

```diff
@@ -77,14 +77,19 @@ type Snapshot struct{ data C.StartupData }
 
 func newSnapshot(data C.StartupData) *Snapshot {
 	s := &Snapshot{data}
-	runtime.SetFinalizer(s, (*Snapshot).release)
```

Comment:
Reverted some of my previous changes, I will try to reproduce the race condition I was running into, I very well could have been using Context incorrectly.

I added a couple test cases around finalizers running, and it doesn't look like ctx is ever getting cleaned up. The Isolate is getting cleaned up, but only if there are no Contexts that were created from that Isolate, I just added a test for an isolate that creates a context to illustrate this. I can add some around values, but ultimately those won't get cleaned up either because the context is holding a pointer to the value(same thing is happening with isolates). 

The reason context is getting cleaned up is because of the [map](https://github.com/augustoroman/v8/blob/master/v8.go#L388) that stores the ctx so we can look up a golang function when it is called from javascript.

Also, from the limited research I have done, and based on your comment, it doesn't sound like a good idea to depend on the finalizers. 

What do you think about exposing the release functions? We could also remove the map of contexts, but I don't know if it is possible because of CGO limitations. This library looks like they don't use a map to keep track of the context: https://github.com/kingland/go-v8, but the last commit was in 2015 and CGO wasn't as strict back then.